libs/newsletter-workflow/src/lib/executeCreate.ts

```diff
@@ -24,8 +25,10 @@ export const executeCreate: AsyncExecution = async (
 
 	const listId = stepData.formData ? stepData.formData['listId'] : undefined;
 	if (!listId) {
+		// TO DO - calculating fields from the Name at this point is not generic
+		// would it be better done somewhere else?
 		const storageResponse = await storageInstance.createDraftNewsletter({
-			...parseResult.data,
+			...calculateFieldsFromName(parseResult.data),
```

Comment:
With the refactor of `calculateFieldsFromName` suggested above, we could change this part to be somethign like this:

```typescript
		// if the form has a non-empty string for a name, use it to calculate the derived fields
		const derivedFields =
			typeof parseResult.data.name === 'string' && !!parseResult.data.name
				? calculateFieldsFromName(parseResult.data.name)
				: {};

		const storageResponse = await storageInstance.createDraftNewsletter({
			...parseResult.data,
			...derivedFields,
			listId: undefined,
		});

```